lib/chains/videoChain.js

```diff
@@ -1,8 +1,235 @@
-const {element} = require('./elementChain');
+const {isNil} = require('ramda');
 
-const videoChain = () => element({video: true});
+// Import utils
+const makeChain = require('../utils/makeChain');
+
+// Import chain composers
+const {
+	notComposer,
+	existComposer,
+	matchComposer,
+	matchRepoComposer,
+	matchJSComposer,
+	// matchBrightScriptComposer,
+	timeoutComposer,
+	makeToStringComposer,
+	makeThenComposer,
+	abandonComposer,
+	cloneComposer,
+	assertComposer,
+	gettersComposer,
+	makeToJSONComposer,
+	isPlayingComposer,
+	isStoppedComposer,
+	isPausedComposer,
+	visibleComposer,
+} = require('../composers');
+const {
+	element,
+	elementExist,
+	elementNotExist,
+	elementVisible,
+	elementMatchProps,
+	elementMatchJS,
+	elementNotMatchJS,
+	elementMatchBRS,
+	elementNotMatchBRS,
+} = require('../texts');
+const {SUBJ_COMPARATOR} = require('../constants/comparator');
+const {
+	processServerResponse,
+	getRequestType,
+	applyTimeout,
+	applyNegation,
+} = require('../utils/socketChainHelper');
+const {
+	SUBJ_COMPARATOR: SUBJ_COMPARATOR_M,
+	VALUE: VALUE_M,
+} = require('../mappings');
+const {VALUE} = require('../constants/element');
+const logger = require('../utils/logger');
+const getOpType = require('../utils/opType').getOpType;
+const {placeholdEmpty} = require('../utils/stringUtils');
+
+/**
+ * Returns human readable description of the video chain.
+ * @param data - chain data
+ * @param nameOnly - if true only element name will be returned (like: {"apiId" : "boo"})
+ * @returns {*}
+ */
+const toString = (data, nameOnly = false) => {
+	const elName = 'video';
+
+	if (nameOnly)
+		return elName;
+
+	let out = element(elName);
+
+	if (data.comparator) {
+		switch (data.comparator.type) {
+			case SUBJ_COMPARATOR.VISIBLE:
+				out = elementVisible(elName);
+				break;
+			case SUBJ_COMPARATOR.EXIST:
+				out = (data.isNegated ? elementNotExist : elementExist)(elName);
+				break;
+			case SUBJ_COMPARATOR.MATCH_JS:
+				out = (data.isNegated ? elementNotMatchJS : elementMatchJS)(elName, data.comparator.val);
+				break;
+			case SUBJ_COMPARATOR.MATCH_BRS:
+				out = (data.isNegated ? elementNotMatchBRS : elementMatchBRS)(elName, data.comparator.val);
+				break;
+			default: {
+				const matches = data.comparator.props.map(
+					one => `  ${one.name} ${one.type} ${placeholdEmpty(VALUE_M[one.val] || one.val)}`
+				);
+
+				out = elementMatchProps(elName, '\n' + matches.join(',\n'));
+			}
+		}
+	}
+
+	return out;
+};
+const beforeSendMsg = (data) => logger.log(getOpType(data), toString(data));
+
+const toJSON = data => {
+	const type = getRequestType(data);
+	const socketMessage = {type};
+	const subject = {
+		type: 'element',
```

Comment:
Perhaps you could also update subject definition according to https://mineus.atlassian.net/browse/SUIT-13143 - feel free to pull that ticket into sprint.
Also, should keep in mind PlayStation support release. We'll need to prepare similar changes for psVideo.

Reply:
Sure, I will take this ticket